modules/registrars/openprovider/Controllers/System/ConfigController.php

```diff
@@ -179,7 +181,17 @@ protected function checkCredentials($configarray, $params)
             Configuration::get('api_url') :
             Configuration::get('api_url_cte');
 
-        if ($this->session->has('AUTH_TOKEN')) {
+
+        $tokenResult = null;
+
+        if (Capsule::schema()->hasTable('reseller_tokens')) {
+            $tokenResult = Capsule::table('reseller_tokens')->where('username', $params['Username'])->orderBy('created_at', 'desc')->first();
+        }
+
+        $expireTime = is_null($tokenResult) ? false : new Carbon($tokenResult->expire_at);
+        $isExpired = $expireTime ? Carbon::now()->diffInSeconds($expireTime, false) < 0 : true;
```

Comment:
I have an idea how to do it better:
$isAlive = $expireTime && Carbon::now()->diffInSeconds($expireTime, false) < 0;
if ($tokenResult && $isAlive) {
...